src/database.ts

```diff
@@ -1130,6 +1130,8 @@ class Database extends common.GrpcServiceObject {
       } catch (e) {
         setSpanErrorAndException(span, e as Error);
         this.emit('error', e);
+      } finally {
```

Comment:
Earlier we also added `span.end()` in all the places where `_releaseOnEnd` was invoked. So if we are adding it here, we should remove it from all the calling methods?

Reply:
Not really. See for example this code pattern 
<img width="611" alt="Screenshot 2024-10-22 at 12 15 51 AM" src="https://github.com/user-attachments/assets/03e3b322-6707-4fa0-9fd1-f3e698a08321">
that code has at least 3 different alternate code paths and they can be confusing for customers if customers if for example they don't end the transaction, spans are going to pile up without some sort of garbage collection; if there is an error encountered but it is returned along with the transaction in the callback the customer sees the error and exits the callback. I added an extra guard to it `if (span.isRecording())`